src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -363,12 +364,15 @@ public void beforeBulk(long executionId, BulkRequest request) {
       public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
         List<DocWriteRequest<?>> requests = request.requests();
 
+        int idx = 0;
         for (BulkItemResponse bulkItemResponse : response) {
-          handleResponse(bulkItemResponse, executionId);
+          boolean failed = handleResponse(bulkItemResponse, executionId);
```

Comment:
nit: I feel like this could be done more elegantly by pairing up the request and responses before iterating through them.

Reply:
Yeah, I considered using guava's `Streams.forEachPair` but it doesn't have quite the same behavior, and also it's just a couple of lines of difference.